### PR TITLE
Turbo over Vector of DateTimes

### PR DESCRIPTION
**Pull request by hayesall**
- Add example documentation for DateTime vectors
  - Based on #339
  - Problem Overview
  - Benchmarks showing 3x speedup with constant memory requirements
- Add new example doc to Documenter build step
- Add new example doc to "Manual Outline" on home page

Image previewing below.

---

image

### CODE REVIEW

**Comment by chriselrod** [comment]
I think this looks good.
I'd just like to see an `@inbounds` on the sequential version and updated benchmarks.
If you don't have AVX512, I don't expect to see too much of an improvement from `@inbounds` because LLVM does not handle the conversion from `Int64` to `Float64` well.

File: docs/src/examples/datetime_arrays.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+```julia
+using Dates
+
+function scale_timeseries_sequential(data::Vector{Dates.DateTime})
+  out = similar(data, Float64)
+  ϕ = (data[lastindex(data)] - data[1]).value
+
+  for i ∈ eachindex(data)
+      out[i] = (data[i] - data[1]).value / ϕ
```
[the remark this answers is not in the document]
Comment (hayesall) [reply docs/src/examples/datetime_arrays.md:57]: Done :heavy_check_mark: 

Took it out for comparison and forgot to add it back in. The sample inputs were still small enough it didn't make a difference.

**Comment by hayesall** [comment]
Benchmarks ran on my i7-4770k, which (I'm pretty sure) has AVX instructions.

By the way, what did you use for generating the GFLOPS vs. size plots ([like this one](https://juliasimd.github.io/LoopVectorization.jl/stable/examples/matrix_vector_ops/))?

**Comment by chriselrod** [comment]
> Benchmarks ran on my i7-4770k, which (I'm pretty sure) has AVX instructions.

Yes, AVX, FMA, and AVX2.

> By the way, what did you use for generating the GFLOPS vs. size plots ([like this one](https://juliasimd.github.io/LoopVectorization.jl/stable/examples/matrix_vector_ops/))?

The code is in the [benchmark](https://github.com/JuliaSIMD/LoopVectorization.jl/blob/master/benchmark/) directory. The script to run the benchmarks is in driver.jl
plotbenchmark.jl has the code calling `plot` and specifying colors to be consistant.

For actually counting flops, I just did that analytically.
Integer subtraction technically isn't a floating point operation, but you could just use `2*length(data)` or `3*length(data)` and call it something like "OPS" (leaving off the "F").

The code running the benchmarks and recording the flops is here:
https://github.com/JuliaSIMD/LoopVectorization.jl/blob/master/benchmark/benchmarkflops.jl
but it is a mess / not something I'd recommend emulating.

More generally, you could measure GFLOPS by removing `@turbo` and using https://github.com/triscale-innov/GFlops.jl (noting again that it probably won't count things like integer ops).

**Comment by chriselrod** [comment]
You're welcome to make another PR to add a GFLOPS plot, but I figured there was no reason to hold this one up.

Thanks for the contribution!

**Pull request merged**